include/allegro5/allegro_x.h

```diff
@@ -0,0 +1,38 @@
+/*         ______   ___    ___
+ *        /\  _  \ /\_ \  /\_ \
+ *        \ \ \L\ \\//\ \ \//\ \      __     __   _ __   ___
+ *         \ \  __ \ \ \ \  \ \ \   /'__`\ /'_ `\/\`'__\/ __`\
+ *          \ \ \/\ \ \_\ \_ \_\ \_/\  __//\ \L\ \ \ \//\ \L\ \
+ *           \ \_\ \_\/\____\/\____\ \____\ \____ \ \_\\ \____/
+ *            \/_/\/_/\/____/\/____/\/____/\/___L\ \/_/ \/___/
+ *                                           /\____/
+ *                                           \_/__/
+ *
+ *      Header file for X specific functionality.
+ *
+ *      By Trent Gamblin.
+ *
+ */
+
+#ifndef __al_included_allegro5_allegro_x_h
+#define __al_included_allegro5_allegro_x_h
+
+#include <X11/Xlib.h>
+
+#ifdef __cplusplus
+   extern "C" {
+#endif
+
+/*
+ *  Public X-related API
+ */
+AL_FUNC(bool, al_get_x_window_handle, (ALLEGRO_DISPLAY *display, XID *out));
```

Comment:
I think just returning the XID is better. You can `ASSERT` that display is not NULL.